### PR TITLE
[W05][W09-4]Sim Jia Zhi

### PR DESCRIPTION
**Pull request by simjiazhi**
Added PersonSorter Comparator to sort records by name
Added case-insensitive Sort command
Updated User Guide to include section for Sort command 
Added JUnit and CLI test cases for Sort command

### CODE REVIEW

**Comment by Xue-Chenyang** [comment]
A good and concise application making use of Java's built in sorting function of collections.
A well attempt to add a new feature while maintaining the overall structure of code and OOP principles.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -151,6 +152,11 @@ Views all details of the 2nd person in the address book.
 `viewall 1` +
 Views all details of the 1st person in the results of the `find` command.
 
+== Sorting entries: `sort`
```
Comment (okkhoy) [inline docs/UserGuide.adoc:155]: 👍 good to see you have updated the user docs

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.List;
+
+public class SortCommand extends Command {
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/SortCommand.java:7]: this class and its non trivial methods need header comments.

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -68,6 +71,15 @@ public UniquePersonList getAllPersons() {
         return new UniquePersonList(allPersons);
     }
 
+    /**
+     * Sort the list of Persons using their names
```
Comment (okkhoy) [inline src/seedu/addressbook/data/AddressBook.java:75]: Sort**s** ...

File: src/seedu/addressbook/data/AddressBook.java [hunk 4]
```diff
@@ -68,6 +71,15 @@ public UniquePersonList getAllPersons() {
...
+     * alphabetically
+     */
+    public void sort()
+    {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/AddressBook.java:79]: coding standard violation here.

File: src/seedu/addressbook/parser/Parser.java [hunk 5]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/parser/Parser.java:14]: do not import everything using `*`; import only those necessary classes

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+public class SortCommandTest {
+
+    private AddressBook emptyAddressBook;
+    private AddressBook addressBook;
+    private AddressBook expectedAddressBook;
+
+    @Before
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:19]: 💯 good use of `@Before`

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                Collections.emptySet());
+
+        emptyAddressBook = TestUtil.createAddressBook();
+        addressBook = TestUtil.createAddressBook(johnDoe, janeDoe, davidGrant, samDoe);
+
+        expectedAddressBook = TestUtil.createAddressBook(davidGrant,janeDoe, johnDoe,samDoe);
+    }
+
+    @Test
+    public void testSort(){
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:38]: Take care of the naming convention for test cases.

**Review by okkhoy — COMMENTED** [review]
Overall good effort in updating all aspects of the enhancement.
Do take care of and adhere to the coding standards next time when you write code.

**Comment by okkhoy** [comment]
@simjiazhi 
Some comments added. Please take note and close the pull request.

**Pull request closed without merging**